File: src/IsbInterop.CSharp/Folders/Wrappers/Folder.cs
using IsbInterop.Base.Wrappers;

namespace IsbInterop.Folders.Wrappers
{
  /// <summary>
  /// Обертка над IFolder.
  /// </summary>
  internal class Folder : EdmsObject<IFolderInfo>, IFolder
  {
    /// <summary>
    /// Получить информацию об объекте.
    /// </summary>
    /// <returns>Информация об объекте.</returns>
    public override IFolderInfo GetInfo()
    {
      var rcwIFolderInfo = GetRcwObjectInfo();
      return new FolderInfo(rcwIFolderInfo, Scope);
    }

    /// <summary>
    /// Конструктор.
    /// </summary>
    /// <param name="rcwIFolder">СOM-объект IFolder.</param>
    /// <param name="scope">Область видимости.</param>
    public Folder(object rcwIFolder, IScope scope) : base(rcwIFolder, scope) { }
  }
}

File: src/IsbInterop/Folders/Wrappers/Folder.vb
Imports IsbInterop.Base.Wrappers

Namespace Folders.Wrappers

  ''' <summary>
  ''' Обертка над IFolder.
  ''' </summary>
  Friend Class Folder
    Inherits EdmsObject(Of IFolderInfo)
    Implements IFolder

    ''' <summary>
    ''' Информация об объекте.
    ''' </summary>
    Public Overrides ReadOnly Property Info As IFolderInfo
      Get
        Dim rcwIFolderInfo = GetRcwObjectInfo()
        Return New FolderInfo(rcwIFolderInfo, Scope)
      End Get
    End Property

    ''' <summary>
    ''' Конструктор.
    ''' </summary>
    ''' <param name="rcwIFolder">СOM-объект IFolder.</param>
    ''' <param name="scope">Область видимости.</param>
    Public Sub New(rcwIFolder As Object, scope As IScope)
      MyBase.New(rcwIFolder, scope)
    End Sub
  End Class
End Namespace